### PR TITLE
[CBRD-23608] Revise error handling and cleanup when applying tde to files fails

### PR DESCRIPTION
**Pull request by wrlawodms**
http://jira.cubrid.org/browse/CBRD-23608

This fetch includes:
- Fix to cleanup VFIDs if `file_apply_tde_algorithm()` fails (in some cases)
- Revise error handling and missed cleanup OR_CLASS_REPR in `xfile_apply_tde_to_class_files()`

### CODE REVIEW

File: src/storage/btree.c [hunk 1]
```diff
@@ -1973,9 +1973,15 @@ btree_create_overflow_key_file (THREAD_ENTRY * thread_p, BTID_INT * btid)
   error_code = heap_get_class_tde_algorithm (thread_p, &btid->topclass_oid, &tde_algo);
   if (error_code != NO_ERROR)
     {
+      VFID_SET_NULL (&btid->ovfid);
```
Comment (hornetmj) [inline src/storage/btree.c:1976]: The created overflow file has to destroy If you want to set btid->ovfid to NULL. Isn't it?

File: src/storage/btree.c [hunk 2]
```diff
@@ -1973,9 +1973,15 @@ btree_create_overflow_key_file (THREAD_ENTRY * thread_p, BTID_INT * btid)
...
       return error_code;
     }
   error_code = file_apply_tde_algorithm (thread_p, &btid->ovfid, tde_algo);
+  if (error_code != NO_ERROR)
+    {
+      VFID_SET_NULL (&btid->ovfid);
```
Comment (hornetmj) [inline src/storage/btree.c:1982]: The created overflow file has to destroy If you want to set btid->ovfid to NULL. Isn't it?

File: src/storage/btree.c (hunk 1 above)
Comment (hornetmj) [reply src/storage/btree.c:1976]: @wrlawodms I understood 100%.

File: src/query/query_manager.c [hunk 3]
```diff
@@ -2509,8 +2509,9 @@ qmgr_get_new_page (THREAD_ENTRY * thread_p, VPID * vpid_p, QMGR_TEMP_FILE * tfil
 
       if (file_apply_tde_algorithm (thread_p, &tfile_vfid_p->temp_vfid, tde_algo) != NO_ERROR)
 	{
-	  file_temp_retire (thread_p, &tfile_vfid_p->temp_vfid);
 	  ASSERT_ERROR ();
+	  file_temp_retire (thread_p, &tfile_vfid_p->temp_vfid);
```
[the remark this answers is not in the document]
Comment (wrlawodms) [reply src/query/query_manager.c:2513]: Yes, the latter is the case in which the temp-file has been already created, but in the former case, the file is being created.

**Review by hornetmj — APPROVED** [review]

**Review by JoohoK — APPROVED** [review]

**Pull request merged**